### PR TITLE
[quant][graphmode] Test for per channel graph mode quantization

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29494 [quant][graphmode] Returning axis from calculate_qparams
* **#29493 [quant][graphmode] Test for per channel graph mode quantization**
* #29416 [quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`

Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18580907](https://our.internmc.facebook.com/intern/diff/D18580907)

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -1106,8 +1120,10 @@ def forward(self, x):
                       if x.startswith('_observer_')])
         assert len(dtypes) == 2, 'Expected to have 2 different types of dtype'
 
-    @_tmp_donotuse_dont_inline_everything
-    def test_insert_quant_dequant(self):
+    @given(
+        is_per_channel=st.booleans()
+    )
```
Comment (ZolotukhinM) [inline test/test_jit.py]: Can we replace it with `for is_per_channel in [True, False]` and not pull hypothesis here?

File: test/test_jit.py [hunk 2]
```diff
@@ -1449,76 +1466,83 @@ def forward(self, x):
...
-        for name, M, data in [('linear', QLinear, torch.randn((5, 5), dtype=torch.float)),
-                              ('conv', QConv, torch.randn((1, 3, 24, 24), dtype=torch.float))]:
-            m = torch.jit.script(M())
-            ref_res = get_forward(m._c)(data)
+    @no_deadline
+    @given(
+        is_per_channel=st.booleans()
+    )
+    def test_fold_prepack(self, is_per_channel):
+        for name, M, ref_M, data in [
```
Comment (ZolotukhinM) [inline test/test_jit.py]: Having several ifs on the same condition is not great. How about we split this to several tests rather than iterating through different configurations in a loop in one test and extract the common part to a separate function?

File: test/test_jit.py [hunk 3]
```diff
@@ -36,10 +36,21 @@
 import torch.jit._logging
 import torch.jit.frontend
 import torch.jit.quantized
+from torch.jit._recursive import wrap_cpp_module
 import torch.nn as nn
 import torch.nn.functional as F
 from torch.quantization import QConfig
 from torch.quantization._quantize_script import ConvPackedParams, LinearPackedParams
+from torch.quantization._quantize_script import script_qconfig
```
Comment (ZolotukhinM) [inline test/test_jit.py]: We probably need to move all quantization related tests from this file into a separate file (no need to do it in this PR).

**Review by ZolotukhinM — CHANGES_REQUESTED** [review]

File: test/test_jit.py (hunk 1 above)
Comment (jerryzh168) [reply test/test_jit.py]: isn't this cleaner?

File: test/test_jit.py (hunk 2 above)
Comment (jerryzh168) [reply test/test_jit.py]: most of the code are the same, how about put the if conditions in small functions?

**Comment by jerryzh168** [comment]
moved some if statements to separate functions, please take a look again

**Review by ZolotukhinM — APPROVED** [review]
Thanks, looks good!

File: test/test_jit.py [hunk 4]
```diff
@@ -36,10 +36,21 @@
...
+from torch.quantization import default_observer
+from torch.quantization import default_weight_observer
+from torch.quantization import default_per_channel_weight_observer
+
+from torch.quantization import quantize
+from common_quantization import SingleLayerLinearModel, AnnotatedSingleLayerLinearModel
+from common_quantization import ConvModel, AnnotatedConvModel
+from common_quantization import test_only_eval_fn
```
Comment (jamesr66a) [inline test/test_jit.py]: FYI this import breaks running under pytest because the imported function's name starts with `test_` :p 

I'll send a quick patch to fix
